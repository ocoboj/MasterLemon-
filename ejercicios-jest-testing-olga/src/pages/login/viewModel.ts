import { FieldValidationResult } from 'lc-form-validation';

export interface LoginEntity {
  login: string;
  password: string;
}

export const createEmptyLoginEntity = (): LoginEntity => ({
  login: '',
  password: '',
});

export interface LoginFormErrors {
  login: FieldValidationResult;
  password: FieldValidationResult;
}

export const createEmptyLoginFormErrors = (): LoginFormErrors => ({
  login: new FieldValidationResult(),
  password: new FieldValidationResult(),
});
