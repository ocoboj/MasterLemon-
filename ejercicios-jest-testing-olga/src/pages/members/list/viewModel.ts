export interface Member {
  id: number;
  name: string;
  avatarUrl: string;
}
