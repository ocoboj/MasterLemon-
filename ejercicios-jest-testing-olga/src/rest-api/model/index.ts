export * from './loginEntity';
export * from './member';
