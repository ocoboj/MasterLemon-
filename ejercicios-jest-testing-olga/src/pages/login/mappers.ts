import * as model from '../../rest-api/model';
import * as vm from './viewModel';

export const mapLoginEntityVMToModel = (loginEntity: vm.LoginEntity): model.LoginEntity => (
  Boolean(loginEntity) ?
  {
    ...loginEntity,
  } :
  null
);
