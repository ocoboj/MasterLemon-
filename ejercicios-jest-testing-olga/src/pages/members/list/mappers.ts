import * as model from '../../../rest-api/model';
import * as vm from './viewModel';

export const mapMemberListModelToVM = (members: model.Member[]): vm.Member[] => (
  Array.isArray(members) ?
    members.map(mapMemberModelToVM) :
    []
);

const mapMemberModelToVM = (member: model.Member): vm.Member => ({
  id: member.id,
  name: member.login,
  avatarUrl: member.avatar_url,
});
