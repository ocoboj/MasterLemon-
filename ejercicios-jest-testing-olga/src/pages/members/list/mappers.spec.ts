import * as model from '../../../rest-api/model';
import * as vm from './viewModel';
import {mapMemberListModelToVM} from './mappers';

describe('pages/members/list/mappers specs', () => {
    describe('mapMemberListModelToVm', () => {

        it('should return empty array when passing members equals []', () => {
            const member: model.Member[] = [];
            const result = mapMemberListModelToVM(member);
            expect(result).toEqual([]);
            expect(result).toMatchSnapshot();
        })

        it('should return array when members are equals undefined', () => {
            const member: model.Member[] = undefined;
            const result = mapMemberListModelToVM(member);
            expect(result).toEqual([]);
            expect(result).toMatchSnapshot();
        });
    });

    it('should return array with item when passing memebers if equals array with item', () => {
        const member: model.Member[] = [
            {
                id:3,
                login: 'test login',
                avatar_url: 'test avatar',
            },
        ];
        const result = mapMemberListModelToVM(member);

        const expectRes: vm.Member[] = [
            {
                id: 3,
                name: 'test login',
                avatarUrl: 'test avatar'
            },
        ];
        expect(result).toEqual(expectRes);
        expect(result).toMatchSnapshot();
    });
    
    it('should return array with 3 items when passing members equals array', () => {
        const member: model.Member[] = [
            {
                id:6,
                login: 'test login 6',
                avatar_url: 'test avatar 6'
            },
            {
                id: 8,
                login: 'test login 8',
                avatar_url: 'test avatar 8'
            },
            {
                id: 10,
                login: 'test login 10',
                avatar_url: 'test avatar 10'
            },
          
        ];
        const result = mapMemberListModelToVM(member);

        const expectRes: vm.Member[] = [
            {
                id:6,
                name: 'test login 6',
                avatarUrl: 'test avatar 6'
            },
            {
                id: 8,
                name: 'test login 8',
                avatarUrl: 'test avatar 8'
            },
            {
                id: 10,
                name: 'test login 10',
                avatarUrl: 'test avatar 10'
            },
         
        ];
        expect(result).toEqual(expectRes);
        expect(result).toMatchSnapshot();
    });
    
    it('should return empty array when passing members equal null', () => {
        const member : model.Member [] = null;

        const result = mapMemberListModelToVM(member);

        expect(result).toEqual([]);
        expect(result).toMatchSnapshot();
    });
});