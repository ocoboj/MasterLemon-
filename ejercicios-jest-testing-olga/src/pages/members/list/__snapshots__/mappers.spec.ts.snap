// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/mappers specs mapMemberListModelToVm should return array when members are equals undefined 1`] = `Array []`;

exports[`pages/members/list/mappers specs mapMemberListModelToVm should return empty array when passing members equals [] 1`] = `Array []`;

exports[`pages/members/list/mappers specs should return array with 3 items when passing members equals array 1`] = `
Array [
  Object {
    "avatarUrl": "test avatar 6",
    "id": 6,
    "name": "test login 6",
  },
  Object {
    "avatarUrl": "test avatar 8",
    "id": 8,
    "name": "test login 8",
  },
  Object {
    "avatarUrl": "test avatar 10",
    "id": 10,
    "name": "test login 10",
  },
]
`;

exports[`pages/members/list/mappers specs should return array with item when passing memebers if equals array with item 1`] = `
Array [
  Object {
    "avatarUrl": "test avatar",
    "id": 3,
    "name": "test login",
  },
]
`;

exports[`pages/members/list/mappers specs should return empty array when passing members equal null 1`] = `Array []`;
